packages/wallet-adapter-core-new/src/WalletCoreNew.ts

```diff
@@ -0,0 +1,1020 @@
+
+  // Local private variable to hold the account that is currently connected
+  private _account: AdapterAccountInfo | null = null;
+
+  // JSON configuration for AptosConnect
+  private _dappConfig: DappConfig | undefined;
+
+  // Private array that holds all the Wallets a dapp decided to opt-in to
+  private _optInWallets: ReadonlyArray<AvailableWallets> = [];
+
+  // Local flag to disable the adapter telemetry tool
+  private _disableTelemetry: boolean = false;
+
+  // Google Analytics 4 module
+  private readonly ga4: GA4 | null = null;
+
+  constructor(
+    optInWallets?: ReadonlyArray<AvailableWallets>,
+    dappConfig?: DappConfig,
+    disableTelemetry?: boolean
+  ) {
+    super();
+    this._optInWallets = optInWallets || [];
+    this._dappConfig = dappConfig;
+    this._disableTelemetry = disableTelemetry || false;
+    this._sdkWallets = getSDKWallets(this._dappConfig);
+
+    // If disableTelemetry set to false (by default), start GA4
+    if (!this._disableTelemetry) {
+      this.ga4 = new GA4();
+    }
+    // Strategy to detect AIP-62 standard compatible extension wallets
+    this.fetchExtensionAIP62AptosWallets();
+    // Strategy to detect AIP-62 standard compatible SDK wallets.
+    // We separate the extension and sdk detection process so we dont refetch sdk wallets everytime a new
+    // extension wallet is detected
+    this.fetchSDKAIP62AptosWallets();
+
+    this.appendNotDetectedStandardSupportedWallets();
+  }
+
+  private fetchExtensionAIP62AptosWallets(): void {
+    let { aptosWallets, on } = getAptosWallets();
+    this.setExtensionAIP62Wallets(aptosWallets);
+
+    if (typeof window === "undefined") return;
+    // Adds an event listener for new wallets that get registered after the dapp has been loaded,
+    // receiving an unsubscribe function, which it can later use to remove the listener
+    const that = this;
+    const removeRegisterListener = on("register", function () {
+      let { aptosWallets } = getAptosWallets();
+      that.setExtensionAIP62Wallets(aptosWallets);
+    });
+
+    const removeUnregisterListener = on("unregister", function () {
+      let { aptosWallets } = getAptosWallets();
+      that.setExtensionAIP62Wallets(aptosWallets);
+    });
+  }
+
+  /**
+   * Set AIP-62 extension wallets
+   *
+   * @param extensionwWallets
+   */
+  private setExtensionAIP62Wallets(
+    extensionwWallets: readonly AptosWallet[]
+  ): void {
+    // Twallet SDK fires a register event so the adapter assumes it is an extension wallet
+    // so filter out t wallet, remove it when twallet fixes it
+    const wallets = extensionwWallets.filter(
+      (wallet) => wallet.name !== "Dev T wallet" && wallet.name !== "T wallet"
+    );
+
+    wallets.map((wallet: AptosStandardWallet) => {
+      if (this.excludeWallet(wallet)) {
+        return;
+      }
+
+      // Remove optional duplications in the _all_wallets array
+      this._standard_wallets = this._standard_wallets.filter(
```

Comment:
for some reason, only Petra keeps showing twice on the wallet selector in some cases, so it is needed mostly for it